protocol/msgpack/buffered_encoder_pool.go

```diff
@@ -42,5 +46,8 @@ func (p *bufferedEncoderPool) Get() BufferedEncoder {
 }
 
 func (p *bufferedEncoderPool) Put(encoder BufferedEncoder) {
+	if p.maxBufferCapacity != 0 && cap(encoder.Buffer().Bytes()) > p.maxBufferCapacity {
```

Comment:
Should probably do `encoder.Buffer().Cap()` here? `.Bytes()` returns the a slice whose length is the same as the number of unread bytes so that's probably not what we want.

Also nit: no need to special check `p.maxBufferCapacity`.

Reply:
Ahh, ok, was not aware of that distinction, will update. 

Initially, I had thought of using 0 to indicate no max capacity, will just change the default to be max int so we don't have to make this check.